### DISCUSSION TITLE
[Feature request] dispatch event when error state changes

### DISCUSSION THREAD

**Discussion by thatdudemanguy** [ing-bank/lion#1222] [Open for contributions]
Hello

As an addition to `showsFeedbackForChanged` and `showsFeedbackForErrorChanged` as found in the [validateMixin](https://github.com/ing-bank/lion/blob/master/packages/form-core/src/validate/ValidateMixin.js), it would be nice to include an event that dispatches whenever the error state itself changes, and have it include all active errors.

e.g.:
```html
<lion-input
  .validators=${[new Required]}
  @error-state-changed=${this._errorFoo}
></lion-input>
```

```js
_errorFoo(e) {
   // e.validationerrors
}
```

**Comment by tlouisse** [comment]
Makes sense. We used to have this in the previous version of ValidateMixin.

All needed info can be found in `.validationStates`. Most likely a solution would be smth along the lines of:

```js
updated(changedProps) {
   super.updated(changedProps);
  
  if (changedProps.has('validationStates')) {
     const prevStates = changedProps.get('validationStates');
     Object.entries(this.validationStates).forEach(([type, feedbackObj]) => {
       if (!deepEqual(feedbackObj, prevStates[type])) {
         this.dispatchEvent(new Event(`${type}StateChanged`, { detail: feedbackObj }));
       }
     });
  }
}
```

**Reply by jorenbroekema** [reply]
Agreed. For validation states `JSON.stringify` equality would suffice as well, not sure if we have a deepEqual util or if we wanna pull in a dependency for it

**Reply by tlouisse** [reply]
True, that was the pseudo part :)

**Comment by tlouisse** [comment]
Since error is 'more important' than other feedback types, we could consider an `invalid` prop and event, so we could support a situation like:

```html
<lion-form @invalid-changed="${({currentTarget}) => {
  this._btnDisabled = currentTarget.invalid;
}}">
  <form>
    <lion-input .validators="${[new Required()]}"></lion-input> 
    <lion-button  type="submit"  .disabled="${this._btnDisabled}">Submit</lion-button>
  </form>
</lion-form>
```
